packages/@o3r/schematics/src/rule-factories/add-imports/index.ts

```diff
@@ -26,17 +25,19 @@ export const addImportsRule = (
     context.logger.warn(`No update applied on file because ${filePath} does not exist.`);
     return;
   }
-  const sourceFile = ts.createSourceFile(
-    filePath,
-    tree.readText(filePath),
-    ts.ScriptTarget.ES2020,
-    true
-  );
-  const recorder = tree.beginUpdate(filePath);
-  const changes = imports.reduce((acc: Change[], { importNames, from }) => acc.concat(
-    importNames.map((importName) => insertImport(sourceFile, filePath, importName, from))
-  ), []);
-  applyToUpdateRecorder(recorder, changes);
-  tree.commitUpdate(recorder);
+  imports.forEach(({ importNames, from }) => {
```

Comment:
change made to group import from the same module

Reply:
That doesn't look optimal to use `ts.createSourceFile` for the same file in a loop.
Can't we do all the updates at once?

Reply:
as discussed we need to do the update of the tree for every import that we want to insert else the sourcefile will not contain the addition of the previous iteration